### PR TITLE
added ctor overload, server can now bind to arbitrary local IP address

### PR DESCRIPTION
**Pull request by vukovinski**
As per #2 

Haven't changed the version. Tested on Windows 10 (19042)

### CODE REVIEW

**Comment by FlorianReimold** [comment]
Great idea! Just 3 remarks:
1. Can you fix the indentation size? I use 2-space indentation and want to keep it consistent
2. In your new constructor, the std::string should be passed as a reference
3. Can you add some documentation to the new functions? 

I will merge the pull request once those points are fixed!

**Comment by vukovinski** [comment]
Did change the argument.

I did the 2 space formatting, but haven't seen any changes.

Added docs, haven't even seen them in the source code, until you mentioned them 🤣

File: fineftp-server/src/server_impl.cpp [hunk 1]
```diff
@@ -137,8 +143,25 @@ namespace fineftp
...
 
-  uint16_t FtpServerImpl::getPort()
+  uint16_t FtpServerImpl::getPort() const
   {
     return acceptor_.local_endpoint().port();
   }
+
+  std::string FtpServerImpl::getAddress() const
+  {
+      return acceptor_.local_endpoint().address().to_string();
```
Comment (FlorianReimold) [inline fineftp-server/src/server_impl.cpp]: 4 Space Indentation 😉

File: fineftp-server/src/server_impl.cpp [hunk 2]
```diff
@@ -137,8 +143,25 @@ namespace fineftp
...
+  asio::ip::tcp::endpoint FtpServerImpl::getEndpoint()
+  {
+      if (address_ == no_address_ || address_.empty())
+      {
+          return asio::ip::tcp::endpoint(asio::ip::tcp::v4(), port_);
+      }
+      else
+      {
+          return asio::ip::tcp::endpoint(asio::ip::address::from_string(address_), port_);
+      }
```
Comment (FlorianReimold) [inline fineftp-server/src/server_impl.cpp]: 4 Space Indentation as well 😀

**Review by FlorianReimold — CHANGES_REQUESTED** [review]
Some 4 spaces-indentations have survived! ;)

**Comment by vukovinski** [comment]
done

**Comment by FlorianReimold** [comment]
Formatting looks good now, but you deleted your repository, so i cannot merge the PR any more.

**Pull request closed without merging**